### PR TITLE
Check error return from closing connection

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR addresses https://github.com/kubernetes/kubernetes/pull/85004#discussion_r346041532

**Which issue(s) this PR fixes**:
Fixes #

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/cert/server_inspection.go [hunk 1]
```diff
@@ -43,9 +44,9 @@ func GetClientCANames(apiHost string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer conn.Close()
+	err = conn.Close()
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/util/cert/server_inspection.go]: if we are going to care about the error (and doing so is ok), I think we should return an unusable value from the function.

```go
if err := conn.Close(); err != nil{
   return nil, err
}
```

Comment (tedyu) [reply staging/src/k8s.io/client-go/util/cert/server_inspection.go]: @deads2k 
Your comment has been addressed.

**Comment by tedyu** [comment]
@deads2k 
Let me know if anything else needs to be done.

File: staging/src/k8s.io/client-go/util/cert/server_inspection.go [hunk 2]
```diff
@@ -72,7 +75,9 @@ func GetServingCertificates(apiHost, serverName string) ([]*x509.Certificate, []
 	if err != nil {
 		return nil, nil, err
 	}
-	conn.Close()
+	if err = conn.Close(); err != nil {
+		return nil, nil, fmt.Errorf("failed to close connection : %v", err)
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/util/cert/server_inspection.go]: nit: I prefer not wrapping errors because lose what they are.

**Comment by deads2k** [comment]
optional nit if you get to it

/lgtm
/approve

**Pull request merged**